File: backend/app/schemas/prediction.py
from pydantic import BaseModel, Field


class ClassInfo(BaseModel):
    raw_label: str
    label: str


class PredictionItem(BaseModel):
    label: str
    raw_label: str
    confidence: float = Field(ge=0, le=1)


class WeatherContext(BaseModel):
    risk_level: str
    reason: str
    location: str
    source: str


class PredictionResponse(BaseModel):
    prediction: str
    raw_label: str | None
    confidence: float = Field(ge=0, le=1)
    is_confident: bool
    validation_status: str | None = None
    validation_message: str | None = None
    validation_reasons: list[str] = Field(default_factory=list)
    top_predictions: list[PredictionItem]
    explanation: str
    next_steps: list[str]
    disclaimer: str
    gradcam_image: str | None = None
    lime_image: str | None = None
    weather_context: WeatherContext | None = None

File: frontend/types/prediction.ts
export type PredictionItem = {
  label: string;
  raw_label: string;
  confidence: number;
};

export type PredictionResponse = {
  prediction: string;
  raw_label: string | null;
  confidence: number;
  is_confident: boolean;
  validation_status?: string | null;
  validation_message?: string | null;
  validation_reasons?: string[];
  top_predictions: PredictionItem[];
  explanation: string;
  next_steps: string[];
  disclaimer: string;
  gradcam_image?: string | null;
  lime_image?: string | null;
  weather_context?: {
    risk_level: string;
    reason: string;
    location: string;
    source: string;
  } | null;
};
